com.github.mtkennerly.ludusavi.yaml

```diff
@@ -0,0 +1,48 @@
+app-id: com.github.mtkennerly.ludusavi
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+build-options:
+  append-path: /usr/lib/sdk/rust-stable/bin
+  env:
+    CARGO_HOME: /run/build/ludusavi/cargo
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  ## Needed for custom paths and restoring saves
+  - --filesystem=host
```

Comment:
Given that it uses [specific locations](https://github.com/mtkennerly/ludusavi/blob/743e524d4337b44851f2567a56f01c04ccd90b49/src/config.rs#L273) to find saves, I don't think host access is warranted here, especially combined with network access.

Reply:
This program also restores gamesaves, which could be located anywere, including external drives, so i dont see how host permission can be avoided without breaking this core functianality.

Reply:
reduced the permissions.

Reply:
@barthalion That function (`find_missing_roots`) is only a set of default launcher install directories, so that Ludusavi can find your installation of the Steam/Epic/etc store clients. But these are just some defaults, and you can add any other folders you like, including custom Wine prefix directories, which could be anywhere.

When it comes to finding the actual game saves, Ludusavi downloads and uses the data from this manifest file, which is updated every 24 hours: https://github.com/mtkennerly/ludusavi-manifest/blob/master/data/manifest.yaml

The saves could technically be anywhere. For example, any save file that depends on `$XDG_DATA_HOME` (`<xdgData>` in the manifest) could be in a surprising location if the user has customized that environment variable. When restoring the backed up saves, users are also able to redirect the saves to another location (e.g., the file was backed up from `/foo/...` but can be restored to `/bar/foo/...`).

Reply:
One example of a surprising save location: [According to PCGamingWiki](https://www.pcgamingwiki.com/wiki/NetHack#Save_game_data_location), Nethack stores its saves in `/var/games/nethack/save/`. Outliers like this are why Ludusavi needs broad file system access.